File: server/utilities.py
from gradio_client import Client, handle_file
from PIL import Image


def predict(path):
    client = Client("rwankhalifa/Fruits_Class")
    result = client.predict(
            image=handle_file(path),
            api_name="/predict"
    )
    return result

from gradio_client import Client, handle_file

def lstm_api(path, temperature):
    client = Client("AlaaAbbas/banana1")
    result = client.predict(
            image=handle_file(path),
            temperature=temperature,
            api_name="/predict"
    )    
    return Image.open(result).convert("RGB")

File: client/src/lib/utilities.js
import { Client, handle_file } from "@gradio/client";

async function predict(path) {
    const client = await Client.connect("rwankhalifa/Fruits_Class");
    const result = await client.predict("/predict",{
        image: handle_file(path),
    });
    return result.data[0];
}

async function HFApi(path, temperature) {
    const client = await Client.connect("AlaaAbbas/banana1");
    const result = await client.predict("/predict",{
        image: handle_file(path),
        temperature: temperature,
        
    });
    return result.data[0].url;
}

export { predict, HFApi };